File: data/python/tests/test_Pydantic_conint.py
import unittest
from function_Pydantic_conint import ProductValidator

class TestCreateProduct(unittest.TestCase):
    
    def test_valid_product(self):
        validator = ProductValidator()
        result = validator.create_product(500, 19.99)
        self.assertEqual(result, "stock=500 price=19.99")

    def test_zero_stock(self):
        validator = ProductValidator()
        result = validator.create_product(0, 5.00)
        self.assertEqual(result, "stock=0 price=5.0")

    def test_stock_at_upper_limit(self):
        validator = ProductValidator()
        result = validator.create_product(1000, 15.00)
        self.assertEqual(result, "stock=1000 price=15.0")

    def test_stock_at_lower_limit(self):
        validator = ProductValidator()
        result = validator.create_product(0, 20.00)
        self.assertEqual(result, "stock=0 price=20.0")

    def test_high_price(self):
        validator = ProductValidator()
        result = validator.create_product(100, 9999.99)
        self.assertEqual(result, "stock=100 price=9999.99")

if __name__ == "__main__":
    unittest.main()

File: data/cpp/tests/test_Pydantic_conint.cpp
#include <gtest/gtest.h>
#include "../src/function_Pydantic_conint.cpp"

class TestCreateProduct : public ::testing::Test {
protected:
    void SetUp() override {
        validator = new ProductValidator();
    }

    void TearDown() override {
        delete validator;
    }
    
    ProductValidator* validator;
};

TEST_F(TestCreateProduct, ValidProduct) {
    std::string result = validator->create_product(500, 19.99);
    EXPECT_EQ(result, "stock=500 price=19.99");
}

TEST_F(TestCreateProduct, ZeroStock) {
    std::string result = validator->create_product(0, 5.00);
    EXPECT_EQ(result, "stock=0 price=5");
}

TEST_F(TestCreateProduct, StockAtUpperLimit) {
    std::string result = validator->create_product(1000, 15.00);
    EXPECT_EQ(result, "stock=1000 price=15");
}

TEST_F(TestCreateProduct, StockAtLowerLimit) {
    std::string result = validator->create_product(0, 20.00);
    EXPECT_EQ(result, "stock=0 price=20");
}

TEST_F(TestCreateProduct, HighPrice) {
    std::string result = validator->create_product(100, 9999.99);
    EXPECT_EQ(result, "stock=100 price=9999.99");
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}